### ISSUE TITLE
No object detection during training single class!

### ISSUE THREAD

**Issue by yatinsd** [AlexeyAB/darknet#4807]
Im trying to train a single class drone using a dataset of around 2600 images, but during training it is not able to detect the object. Dont know why it shows nan for object all the time even when average loss dropped from 2000 to 0.44 in about 890 iterations. I'm attaching the screenshot for the same!
![Screenshot from 2020-02-06 01-28-49](https://user-images.githubusercontent.com/37011467/73898413-2c557300-4881-11ea-8250-55f069f34ca2.png)

**Comment by PaserSRL** [comment]
I'm quite sure your configuration is wrong, please attache *.names *.data and *.cfg file

**Comment by yatinsd** [comment]
[net]
# Testing
batch=64
subdivisions=16
# Training
# batch=64
# subdivisions=2
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 2000
policy=steps
steps=1600,1800
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=18
activation=linear



[yolo]
mask = 3,4,5
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=1
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=18
activation=linear

[yolo]
mask = 1,2,3
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=1
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

**Comment by yatinsd** [comment]
**OBJ.DATA**
classes= 1
train  = /home/dl2/darknet/build/darknet/x64/data/train.txt
valid  = /home/dl2/darknet/build/darknet/x64/data/test.txt
names = /home/dl2/darknet/build/darknet/x64/data/obj.names
backup = /home/dl2/darknet/backup

**Comment by PaserSRL** [comment]
in all labels of your dataset, has drone index 0?
"**0** 0.547190 0.704386 0.108820 0.204102" <- example

**Comment by yatinsd** [comment]
yes this issue is not going! im thinking of applying a negative database, i have increased the resolution as well to 544*544, but still same! Do you think that can help?
